### PR TITLE
Services UI

### PR DESCRIPTION
**Pull request by lieblichaaron**
## ℹ️ Change description:
Nothing here

<details><summary><sub>templates<sub></summary>
<p>
  
<sub>[templates: [bug](?expand=1&template=bug.md), [feature](?expand=1&template=feature.md)]</sub>

</p>
</details>

### CODE REVIEW

File: examples/client/Locomotion/App.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import React from 'react';
+import { LogBox } from 'react-native';
 import LocomotionRouter from './src/LocomotionRouter';
 // import de from './src/I18n/en.json';
+LogBox.ignoreAllLogs();
```
Comment (avivbarniv) [inline examples/client/Locomotion/App.js]: remove

File: examples/client/Locomotion/src/Components/SvgIcon/index.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ const SvgIcon = ({
     onPress={onPress}
     disabled={!onPress}
     margin={margin}
+    noBg
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/SvgIcon/index.js]: What is this flag?

File: examples/client/Locomotion/src/Components/Tag/styled.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import styled from 'styled-components/native';
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/Tag/styled.js]: change to styled-components
and use styled(View)

File: examples/client/Locomotion/src/context/newRideContext/api.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import network from '../../services/network';
+
+export const createServiceEstimations = async (stopPoints) => {
+  const { data } = await network.post('api/v1/services/service-estimations', { stopPoints });
+  return data;
+};
+
+export const getServices = async () => {
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/context/newRideContext/api.js:8]: Why do you need to get the services?

File: examples/client/Locomotion/src/context/newRideContext/index.js [hunk 5]
```diff
@@ -62,7 +64,33 @@ const RidePageContextProvider = ({ navigation, children }) => {
   const [searchResults, setSearchResults] = useState(null);
   const [isReadyForSubmit, setIsReadyForSubmit] = useState(false);
   const [historyResults, setHistoryResults] = useState([]);
-
+  const [serviceEstimations, setServiceEstimations] = useState(null);
+
+  const formatEstimations = (services, estimations) => services.map((service) => {
+    const estimationForService = estimations.find(estimation => estimation.serviceId === service.id);
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/context/newRideContext/index.js]: Please create a map of {
[estimation.serviceId]: esstimation
} before the services.map And then access it via the map.
better runtime

File: examples/client/Locomotion/src/context/newRideContext/index.js [hunk 6]
```diff
@@ -62,7 +64,33 @@ const RidePageContextProvider = ({ navigation, children }) => {
...
+    const estimationResult = estimationForService && estimationForService.results[0];
+    return {
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/context/newRideContext/index.js]: extract it to format function
return formatEstimationResult()

File: examples/client/Locomotion/src/context/newRideContext/index.js [hunk 7]
```diff
@@ -62,7 +64,33 @@ const RidePageContextProvider = ({ navigation, children }) => {
...
+      price: (estimationResult || {}).priceAmount,
+      availableSeats: service.maxPassengers || 4,
+      tag: null,
+      iconUrl: service.icon,
+      description: service.displayDescription,
+    };
+  });
+
+  const getServiceEstimations = async () => {
+    const formattedStopPoints = requestStopPoints.map(sp => ({
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/context/newRideContext/index.js]: extract to function

File: examples/client/Locomotion/src/context/newRideContext/index.js [hunk 8]
```diff
@@ -62,7 +64,33 @@ const RidePageContextProvider = ({ navigation, children }) => {
...
+      type: sp.type,
+      lat: sp.location.lat,
+      lng: sp.location.lng,
+    }));
+    const estimations = await createServiceEstimations(formattedStopPoints);
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/context/newRideContext/index.js]: Not sure why you need the services, maybe return them as part of the payload of estimations
But If you do need it, use promise.all

File: examples/client/Locomotion/src/context/user/index.tsx [hunk 9]
```diff
@@ -63,8 +72,8 @@ const UserContextProvider = ({ children }: { children: any }) => {
     getUserFromStorage();
   }, []);
 
-  const verifyEmail = async (userId: string) => {
-    await sendEmailVerification(userId);
+  const verifyEmail = async () => {
+    await sendEmailVerification(user && user.id);
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/context/user/index.tsx]: you can remove, backend takes it from req.user.id

File: examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/ServiceOptions/ServiceCard/index.js [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+      container: {
+        borderWidth: 1,
+        borderStyle: 'solid',
+        borderColor: theme.primaryColor,
+      },
+      textColor: theme.primaryColor,
+    },
+  };
+  return (
+    <CardContainer theme={theme} selected={selected} noBg>
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/ServiceOptions/ServiceCard/index.js]: What is noBg?

File: examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/ServiceOptions/ServiceCard/index.js [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+            <Circle />
+            <Eta>
+              {timeUntilArrival}
+            </Eta>
+          </TimeDetails>
+          <Capacity>
+            <AvailableSeats>
+              {service.availableSeats}
+            </AvailableSeats>
+            <SvgIcon Svg={Seat} width="15px" height="15px" />
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/ServiceOptions/ServiceCard/index.js]: height and width are now numbers

**Review by avivbarniv — COMMENTED** [review]
See the comments above

File: examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/ServiceOptions/ServiceCard/index.js (hunk 10 above)
Comment (lieblichaaron) [reply examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/ServiceOptions/ServiceCard/index.js]: the button component has a default blue background, so to remove it you add the flag. probably not the best way to do that

File: examples/client/Locomotion/src/context/newRideContext/index.js (hunk 8 above)
Comment (lieblichaaron) [reply examples/client/Locomotion/src/context/newRideContext/index.js]: it just seemed out of the scope of the estimations api to return all the demand sources services also, so i take the services and the estimations that were available and i create the service object needed for the ui

File: examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/ServiceOptions/ServiceCard/index.js (hunk 10 above)
Comment (avivbarniv) [reply examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/ServiceOptions/ServiceCard/index.js]: It's fine for now, but call the flag noBackground

Comment (lieblichaaron) [reply examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/ServiceOptions/ServiceCard/index.js]: ok, but there are a bunch of places that use it

File: examples/client/Locomotion/src/Components/Tag/styled.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import styled from 'styled-components';
+import { FONT_SIZES, FONT_WEIGHTS } from '../../context/theme';
+
+export const Container = styled.View`
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/Tag/styled.js:4]: use styled(View)

File: examples/client/Locomotion/src/Components/Tag/styled.js [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+border-radius: 8px;
+padding: 2px 8px;
+`;
+
+export const TagText = styled.Text`
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/Tag/styled.js:9]: use styled(Text)

File: examples/client/Locomotion/src/context/newRideContext/services.js [hunk 14]
```diff
@@ -0,0 +1,24 @@
+export const TAG_OPTIONS = {
+  FASTEST: 'Fastest',
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/context/newRideContext/services.js]: i18n

**Review by avivbarniv — APPROVED** [review]

**Pull request merged**